### ISSUE TITLE
Introduce semantic versions in Sentry to allow error resolution

### ISSUE THREAD

**Issue by sebathomas** [magma/magma#10827]
# Introduce semantic versions in Sentry to allow error resolution

## Problem

Sentry is an error monitoring solution that collects crashes and error logs from running AGW services. After these errors have been fixed, it is possible to resolve them within the Sentry UI for a given version. Sentry will always consider the version if new reports of the same error come in, and only show the error again as a regression if it occurs for a higher version than the one that resolved it.

For the resolve / regression flow to work, Sentry needs to know the correct order of the versions. Unfortunately, it cannot parse the current versioning scheme of Magma so the versions are ordered by the timestamp when they were first seen. This means all resolved errors that were not backported will be reported as a regression as soon as a new release from an older branch (currently 1.6.1) starts talking to Sentry.

## Solution

Sentry supports semantic release versions if they are given in the [correct format](https://docs.sentry.io/platforms/python/configuration/releases/#bind-the-version). For magma, we propose changing the release version given in the Sentry SDK to the following format:
```bash
VERSION="1.7.0"; echo "magma@${VERSION}.$(git rev-list --count HEAD)-$(git log --pretty=format:'%h' -n 1)"
# magma@version.commit_count-commit_hash
# Example: magma@1.7.0.9829-ae3224e0f
# Sentry versions: major: 1, minor: 7, bugfix: 0, revision: 9829
```
Setting this version instead of the current version (`version-commit_hash`) allows Sentry to understand the order of releases correctly. Using the commit count as revision version guarantees that a later commit in the main branch will be treated as a later release independent of when it was deployed.

Versions using the scheme are shown in Sentry like this:
![releases](https://user-images.githubusercontent.com/14236667/145837891-3427a9e7-8ef1-4566-b9b3-efbb5c7dab48.png)

## Open point

Currently the CI build process sends release dumps to Sentry that follow a third version scheme `version-timestamp-commit_hash` that is incompatible with the versions used in the SDK. It might be useful to use the same version scheme everywhere.

**Comment by ssanadhya** [comment]
@sebathomas, is this proposal still active? Can you present it in the next TSC meeting? If yes, I can add it to the [Mar 28 agenda](https://wiki.magmacore.org/pages/resumedraft.action?draftId=2366682&draftShareId=6e21b3b0-dcac-4944-80cd-f077eb7e9a09&) .

**Comment by sebathomas** [comment]
@ssanadhya This was already implemented in #10831.
